src/Dropdown/Dropdown.js

```diff
@@ -0,0 +1,258 @@
+  /** For use with controlled components, in which the app manages Dropdown state */
+  isOpen?: boolean,
+  /** Plugins used to alter behavior. See [PopperJS docs](https://popper.js.org/popper-documentation.html#modifiers) for options. */
+  modifiers?: Object,
+  /** Called when Dropdown is closed */
+  onClose?: (event: Object) => void,
+  /** Called when Dropdown is opened */
+  onOpen?: (event: Object) => void,
+  /** Placement of the Dropdown menu */
+  placement?:
+    | 'bottom-end'
+    | 'bottom-start'
+    | 'left-end'
+    | 'left-start'
+    | 'right-end'
+    | 'right-start'
+    | 'top-end'
+    | 'top-start',
+  /** Focus trigger after selecting an item */
+  restoreFocus?: boolean,
+  /** Display a wider Dropdown menu */
+  wide?: boolean
+};
+
+type State = {
+  highlightedIndex: null | number,
+  isOpen?: boolean
+};
+
+/**
+ * Dropdown provides a list of actions to the user.
+ */
+export default class Dropdown extends Component {
+  static defaultProps = {
+    placement: 'bottom-start',
+    restoreFocus: true
+  };
+
+  props: Props;
+
+  state: State = {
+    highlightedIndex: null,
+    isOpen: Boolean(this.props.defaultIsOpen)
+  };
+
+  _isMounted: boolean = false;
+
+  dropdownTrigger: React$Component<*, *, *>;
+
+  id: string = `dropdown-${generateId()}`;
+
+  items: Array<Item>;
+
+  selectedItemId: ?string;
+
+  componentDidMount() {
+    this._isMounted = true;
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
+
+  render() {
+    const {
+      children,
+      data,
+      modifiers,
+      placement,
+      wide,
+      ...restProps
+    } = this.props;
+
+    const { isOpen } = this.isControlled() ? this.props : this.state;
+
+    if (isOpen) {
+      this.items = this.getItems();
+
+      this.selectedItemId =
+        this.state.highlightedIndex === null
+          ? undefined
+          : `${this.id}-menuItem-${this.state.highlightedIndex}`;
+    }
+
+    const dropdownTriggerProps = {
+      'aria-activedescendant': this.selectedItemId,
+      'aria-haspopup': true,
+      children,
+      isOpen,
+      onKeyDown: this.onTriggerKeyDown
+    };
+
+    const dropdownContentProps = {
+      data,
+      getItemProps: this.getItemProps,
+      modifiers,
+      placement,
+      wide
+    };
+
+    const rootProps = {
+      id: this.id,
+      restoreFocus: false,
+      ...restProps,
+      autoFocus: false,
+      content: <DropdownContent {...dropdownContentProps} />,
+      isOpen,
+      onClose: this.onClose,
+      onOpen: this.onOpen,
+      trigger: <DropdownTrigger {...dropdownTriggerProps} />,
+      triggerRef: node => {
+        this.dropdownTrigger = node;
+      },
+      wrapContent: false
+    };
+
+    return (
+      <Root {...rootProps}>
+        {children}
+      </Root>
+    );
+  }
+
+  getItems = () => {
+    let items = [];
+
+    this.props.data.forEach(group => {
+      items = items.concat(group.items.filter(item => !item.divider));
+    });
+
+    return items;
```

Comment:
i think you're doing a reduce here, so this might be a bit clearer as

```
return this.props.data.reduce((accum, group) => {
    return accum.concat(group.items.filter(item => !item.divider));
}, []);
```